### PR TITLE
Support -no-stdlib and -no-reflect kotlinc args

### PR DESCRIPTION
**Pull request by Bencodes**
https://kotlinlang.org/docs/reference/compiler-reference.html#-no-reflect and https://kotlinlang.org/docs/reference/compiler-reference.html#-no-stdlib

### CODE REVIEW

**Comment by Kernald** [comment]
In the same idea as https://github.com/bazelbuild/rules_kotlin/pull/390#discussion_r524530762, I would expose these two flags in a single one with multiple options rather than two booleans, given that `-no-stdlib` implies `no-reflect`. Maybe a `include_stdlibs` with `all` (default), `stdlib` (`-no-reflect`), `none` (`-no-stdlib`)?

**Comment by Bencodes** [comment]
Updated to use string options with `all`, `stdlib`, and `none`.

File: kotlin/internal/jvm/compile.bzl [hunk 1]
```diff
@@ -205,6 +205,10 @@ def _kotlinc_options_provider_to_flags(opts):
         flags.append("-Xallow-jvm-ir-dependencies")
     if opts.x_skip_prerelease_check:
         flags.append("-Xskip-prerelease-check")
+    if opts.include_stdlibs == "stdlib":
+        flags.append("-no-reflect")
```
Comment (restingbull) [inline kotlin/internal/jvm/compile.bzl:209]: Doesn't reflect the include_stdlibs attribute.

**Review by restingbull — CHANGES_REQUESTED** [review]

File: kotlin/internal/jvm/compile.bzl (hunk 1 above)
Comment (Bencodes) [reply kotlin/internal/jvm/compile.bzl:209]: @restingbull mind elaborating?

Comment (Bencodes) [reply kotlin/internal/jvm/compile.bzl:209]: The `-no-reflect` flag strips out the Kotlin reflect library but keeps the standard library. The `-no-stdlib` flag strips both the reflect and standard libraries out. Not passing either of these flags will result in both the standard library and reflect being on the classpath.

When the `include_stdlibs ` is set to `all` we want to just include everything (pass no flags). When `include_stdlibs` is set to `stdlib` we want to pass -no-reflect which just strips the reflect standard libraries but keeps the kotlin standard library. Lastly when we pass `none` we want to just strip everything by passing `-no-stdlib`.

This logic also mirrors what @Kernald mentioned here https://github.com/bazelbuild/rules_kotlin/pull/395#issuecomment-728301116

Comment (restingbull) [reply kotlin/internal/jvm/compile.bzl:209]: Nevermind, I'm going brain dead. I read the code as taking a list no less than 5 times. My apologies.

**Review by restingbull — APPROVED** [review]

**Pull request merged**